### ISSUE TITLE
Clearing scheduled event when Player is synced does not work

### ISSUE THREAD

**Issue by KarolPokojowczyk** [Tonejs/Tone.js#691]
First of all a big thank you for this amazing library.

I am having an issue with clearing the scheduled event from the Transport when the scheduled Player is sync()ed. Without sync() everything works fine.

Here's a an example code:
https://codepen.io/caughtinjoy/pen/KKVNxwo
Please click all the buttons in order to reproduce the problem.
As you can see after clear()ing the event it still plays.

Many thanks.

**Comment by tambien** [comment]
thank you! 

Looking at your code, here's my understanding: you `sync()` the player (which means that all subsequent calls to `start()` will schedule a start event along the Transport timeline). Then you call `start()` on the player from within a scheduled callback. Then you cancel that scheduled callback, but since you called `player.sync().start()` the player has now set a start event along the Transport's timeline. So by cancelling the `schedule` event you didn't remove the synchronized `start()` event from the Transport's event timeline. 

It seems to me that you for your desired behavior, you don't want to use `sync()` when you want to synchronize your own scheduling along the Transport using `Transport.schedule`. alternatively you can use `sync()` and `unsync()` if you'd like to remove the event from the Transport's timeline. 

Hope that helps clarify. i'm open to suggestions on how to make this more clear in the API or docs.

**Comment by KarolPokojowczyk** [comment]
Thank you so much for clarifying this. I guess I was mistaken by the name sync() as I was expecting it to sync it to the timeline and then to use it with conjunction with the scheduled callback and with the player.start(). Which I thought will behave similar to all the examples with triggerAttackRelease, but instead of triggerAttackRelease you would call start().

Now it all makes much more sense :) 

Long story short, I am trying to write a simple DAW-like Players/Samplers synced together, where a user would be able to change a specific already scheduled event. I.e. replace a bass line in the 2nd measure in realtime.

Thanks again.

**Comment by tambien** [comment]
great! glad that was cleared up. 

there is also a `sync()` method with the synths which behaves in a similar way as Player and similarly replaces the need to schedule events inside of `Transport.schedule`
